src/clib/pioc_support.c

```diff
@@ -148,35 +188,43 @@ void pio_log(int severity, const char *fmt, ...)
        many tabs before the message. */
     if (!severity)
     {
-        strcpy(ptr, ERROR_PREFIX);
+        strncpy(ptr, ERROR_PREFIX, (rem_len > 0) ? rem_len : 0);
```

Comment:
Also, I think we should have uniform coding style through out the function (easier to modify the code later). I find it more easy to understand and modify. strcpy is waiting for a bug.